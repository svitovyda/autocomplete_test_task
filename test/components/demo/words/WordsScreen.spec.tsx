import { WordsScreen } from '../../../../src/components/demo/words/WordsScreen';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import * as React from 'react';

describe('WordsScreen', () => {
  it('renders without error', () => {
    const component = render(<WordsScreen />);

    expect(component).toMatchSnapshot();
  });
});
