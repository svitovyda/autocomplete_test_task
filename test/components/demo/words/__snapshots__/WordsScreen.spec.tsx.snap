// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordsScreen renders without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": .c1 {
  margin: 8px 0;
  display: block;
  width: 100%;
}

.c0 {
  max-width: 600px;
  margin: 0 auto;
  padding: 0 10px;
  display: block;
}

.c2 {
  width: 100%;
  margin: 0;
  padding: 0;
  display: block;
  height: 0;
}

.c3 {
  height: 50px;
  width: 50px;
  margin: 5px auto;
}

.c4 {
  position: relative;
  top: 0;
  left: 0;
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 3px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: LtRVX 2s linear 0s infinite normal;
  animation: LtRVX 2s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c5 {
  position: relative;
  top: -52px;
  left: 5px;
  width: 43px;
  height: 43px;
  border-radius: 100%;
  border: 2px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: cLNnjC 1s linear 0s infinite normal;
  animation: cLNnjC 1s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c6 {
  position: relative;
  top: -80px;
  left: -22px;
  width: 100px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: kRlIGi 3s linear 0s infinite normal;
  animation: kRlIGi 3s linear 0s infinite normal;
  color: lightgrey;
  font: bold 8px 'Helvetica Neue', Helvetica, Arial, sans-serif;
  opacity: 0;
  text-align: center;
  text-transform: uppercase;
}

<body>
    <div>
      <div
        class="c0"
      >
        <h1>
          Dynamic Data: Words
        </h1>
        <div
          class="c1"
        >
          <input
            placeholder="Start typing to find words..."
            type="text"
            value=""
          />
          <div
            class="c2"
          />
        </div>
        <div
          class="c3"
        >
          <div
            class="c4"
          />
          <div
            class="c5"
          />
          <div
            class="c6"
          >
            loading
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .c1 {
  margin: 8px 0;
  display: block;
  width: 100%;
}

.c0 {
  max-width: 600px;
  margin: 0 auto;
  padding: 0 10px;
  display: block;
}

.c2 {
  width: 100%;
  margin: 0;
  padding: 0;
  display: block;
  height: 0;
}

.c3 {
  height: 50px;
  width: 50px;
  margin: 5px auto;
}

.c4 {
  position: relative;
  top: 0;
  left: 0;
  height: 50px;
  width: 50px;
  border-radius: 100%;
  border: 3px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: LtRVX 2s linear 0s infinite normal;
  animation: LtRVX 2s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c5 {
  position: relative;
  top: -52px;
  left: 5px;
  width: 43px;
  height: 43px;
  border-radius: 100%;
  border: 2px solid transparent;
  border-color: transparent lightgrey transparent lightgrey;
  -webkit-animation: cLNnjC 1s linear 0s infinite normal;
  animation: cLNnjC 1s linear 0s infinite normal;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
}

.c6 {
  position: relative;
  top: -80px;
  left: -22px;
  width: 100px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-animation: kRlIGi 3s linear 0s infinite normal;
  animation: kRlIGi 3s linear 0s infinite normal;
  color: lightgrey;
  font: bold 8px 'Helvetica Neue', Helvetica, Arial, sans-serif;
  opacity: 0;
  text-align: center;
  text-transform: uppercase;
}

<div>
    <div
      class="c0"
    >
      <h1>
        Dynamic Data: Words
      </h1>
      <div
        class="c1"
      >
        <input
          placeholder="Start typing to find words..."
          type="text"
          value=""
        />
        <div
          class="c2"
        />
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
        />
        <div
          class="c5"
        />
        <div
          class="c6"
        >
          loading
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
